### PR TITLE
[17.0][RFC] base_tier_validation: remove unused code

### PR DESCRIPTION
**Pull request by bosd**
remove the code, which was needed for the review widget pre V16.

### CODE REVIEW

**Comment by bosd** [comment]
the function `get_reviews` was no longer called upon in the javascript. After this commit. 
https://github.com/OCA/server-ux/commit/145c5702fc51eaf6622ed38d64ce0dd26b7dc145#diff-5d679fcb388f892b31b8c390404d207ed9b96c806e57db499e13d04443d628e9L33
(or anywhere else afaik)
But it still was unit tested.

The remaining code might need some refactoring :thinking: 

@celm1990 @LoisRForgeFlow 
Can you please review / provide feedback?

File: base_tier_validation/models/res_users.py
Comment (LoisRForgeFlow) [inline base_tier_validation/models/res_users.py]: this

File: base_tier_validation/models/res_users.py
Comment (LoisRForgeFlow) [inline base_tier_validation/models/res_users.py]: and this

**Review by LoisRForgeFlow — COMMENTED** [review]
can you also explain the other changes?

File: base_tier_validation/models/res_users.py
Comment (bosd) [reply base_tier_validation/models/res_users.py]: Had to remove this, to make the tests pass.
I'm not seeing in the code where the values to review_ids are written. 
(expect for the no longer needed lines in the test)
`self.assertTrue(self.test_user_1.get_reviews({"res_ids": review.ids}))`
With no valid ID assigned it broke the tests..

(Filtering on the user ID is done, a couple llines below.)

File: base_tier_validation/models/res_users.py
Comment (bosd) [reply base_tier_validation/models/res_users.py]: Without this. There was an error in the tests. The user is not allowed to acces SO records.

File: base_tier_validation/models/res_users.py
Comment (bosd) [reply base_tier_validation/models/res_users.py]: Maybe some more refactoring is needed on the domain. (Or an alternative function to write `review_ids`) I'm not sure..

**Comment by LoisRForgeFlow** [comment]
@bosd The two changes above that you claimed are needed to fix tests, doesn't look needed as many time has passed and tests of tier validation continue to pass. Can you just remove them and just keep the deletion of unused code? After that, it will be good to merge.

Thanks!

**Comment by bosd** [comment]
@LoisRForgeFlow Done, I removed those fixes.

**Comment by LoisRForgeFlow** [comment]
@bosd test are failing now, can you check? I mean there must be something wrong here, main branch is green

**Comment by bosd** [comment]
@LoisRForgeFlow With the earlier mentiond fixes removed tests are indeed failling.
(Some time has passed since I started working on this. While quickly looking at this, I don't know why it fails now.)
Assuming it was something acces rights related and compute order.

**Comment by LoisRForgeFlow** [comment]
@bosd is you branch rebased on top of the latest 17.0?

**Comment by bosd** [comment]
@LoisRForgeFlow Yes it is.

**Comment by LoisRForgeFlow** [comment]
@bosd  hmmm weird indeed. I guess you will need to have a deeper look when you have time, we cannot merge without the green check.

**Comment by bosd** [comment]
@LoisRForgeFlow Ive included the fixes again. Is it ok to merge it as is?
As I don't expect to have some time available in the near future to dig very deep into this and find the reasoning.

File: base_tier_validation/models/res_users.py
Comment (StefanRijnhart) [reply base_tier_validation/models/res_users.py]: This needs to stay, otherwise the review count will include all users' reviews I fear. The test can be fixed with
```
--- a/base_tier_validation/tests/test_tier_validation.py
+++ b/base_tier_validation/tests/test_tier_validation.py
@@ -417,7 +417,7 @@ class TierTierValidation(CommonTierValidation):
         # Request validation
         review = test_record.with_user(self.test_user_2).request_validation()
         self.assertTrue(review)
-        test_record.invalidate_model()
+        self.env.invalidate_all()
         self.assertTrue(test_record.review_ids)
         # Used by front-end
         count = self.test_user_1.with_user(self.test_user_1).review_user_count()
```

Comment (bosd) [reply base_tier_validation/models/res_users.py]: Thanks, that fixed it!

**Comment by bosd** [comment]
Included the fix, and rebased.

**Review by StefanRijnhart — APPROVED** [review]

File: base_tier_validation/models/res_users.py
Comment (celm1990) [reply base_tier_validation/models/res_users.py]: @bosd I think it is better to use `sudo` or `with_user` in the test. Did you try using these options instead of changing the code to make the test work?

**Review by celm1990 — COMMENTED** [review]
LGTM but I have the same question as @LoisRForgeFlow regarding `sudo`

File: base_tier_validation/models/res_users.py
Comment (bosd) [reply base_tier_validation/models/res_users.py]: It was a long time ago I was working on this. I think I just changed it to make the test work.

Comment (bosd) [reply base_tier_validation/models/res_users.py]: Test pass as well now by using `with_user`. So reverted this change.

**Review by celm1990 — APPROVED** [review]
LGTM

**Review by LoisRForgeFlow — APPROVED** [review]
Thanks

**Pull request merged**
